kernel/src/schema.rs

```diff
@@ -757,6 +786,38 @@ pub trait SchemaTransform {
     }
 }
 
+struct SchemaLeaves {
+    path: Vec<String>,
+    names: Vec<ColumnName>,
+    types: Vec<DataType>,
+}
+impl SchemaLeaves {
+    fn new(own_name: Option<&str>) -> Self {
+        Self {
+            path: own_name.into_iter().map(|s| s.to_string()).collect(),
```

Comment:
so this is just a single-item vec with the own name to start right?

Reply:
Or an empty vec, if no "own" name was provided. If you look at the visitor call sites of `StructType::leaves`, all the visitors involved in log replay have to pass an "own" name, because we're parsing `add.*`, `remove.*`, etc; but inspect-table and scan data visitors are directly visiting the top-level columns of a query result and so they do _not_ have an "own" name for the paths they access.

Reply:
yup got it thanks!